core/trino-main/src/main/java/io/trino/connector/informationschema/InformationSchemaMetadata.java

```diff
@@ -305,6 +305,7 @@ private Set<QualifiedTablePrefix> calculatePrefixesWithTableName(
                     })
                     .filter(objectName -> predicate.isEmpty() || predicate.get().test(asFixedValues(objectName)))
                     .map(QualifiedObjectName::asQualifiedTablePrefix)
+                    .limit(MAX_PREFIXES_COUNT + 1)
```

Comment:
It looks weird to limit before collecting to a set.
Set will remove duplicates, so it's cardinality may be less than original stream's.

Reply:
Fair point. The stream may contain duplicates here because of the `flatMap` operation from line 277 - 279
`.flatMap(prefix -> prefix.getSchemaName()
                            .map(schemaName -> Stream.of(prefix))
                            .orElseGet(() -> listSchemaNames(session)))`

`listSchemaNames(session)` may be called multiples time. I can try rewriting it with for-each loop and end when the count exceeds `MAX_PREFIXED_COUNT`? Or just let it be.

Reply:
@phd3 See https://github.com/trinodb/trino/blob/master/core/trino-main/src/main/java/io/trino/connector/informationschema/InformationSchemaMetadata.java#L224-L230
`tablePrefixes` is computed by `calculatePrefixesWithTableName` in the if conditions in line 224 to 230, if the size of `tablePrefixes` is larger than `MAX_PREFIXES_COUNT` it will be replaced with `defaultPrefixes(catalogName)`.

So we should limit the stream at `MAX_PREFIXES_COUNT + 1` to know if the size of the stream is larger than `MAX_PREFIXES_COUNT`